README.md

```diff
@@ -629,6 +630,114 @@ These are some of the **important** principles to keep in mind when creating and
 
 This section provides some details on more advanced Dojo 2 functionality and configuration that may be required to build more complex widgets and applications.
 
+### Handling Focus
+
+Handling focus is an important aspect in any application and can often be tricky to do correctly. To help with this issue @dojo/widget-core provides a primitive mechanism built into the VDOM system that enables users to focus a virtual dom node once it has been appended to the DOM. This uses a special property called `focus` on the `VNodeProperties` interface that can be passed when using `v()`. The `focus` property is either a `boolean` or a function that returns a `boolean`.
+
+When using passing a function focus will called when `true` is returned without comparing the value of the previous result, however when passing a `boolean` focus will only be applied if the property is `true` and the previous property value was not.
+
+
+```ts
+// Call focus for the node on the only first render
+v('input', { type: 'text', focus: true })
+```
+
+```ts
+// Call focus for the node on every render
+v('input', { type: 'text', focus: () => true) })
+```
+
+This primitive is a base that enables building further abstractions to handle more complex behaviors, one of which is handling focus across the boundaries of encapsulated widgets. The `FocusMixin` is designed to provide support for these scenarios and should be used by widgets that want to provide `focus` to its children or can accept `focus` from a parent widget.
+
+The mixin enhances a widgets API adding `focus` and `shouldFocus` methods, `shouldFocus` checks whether the widget is in a state to perform a focus action and will only return `true` once until the the widget's `focus` method has been called. This `shouldFocus` method is designed to be passed to child widgets or nodes as the value of the `focus` property.
```

Comment:
the `FocusMixin` adds `focus` and `shouldFocus` properties to a widget's API. `shouldFocus` checks if the widget is in a state to perform a focus action and will only return `true` once the widget's `focus` method has been called. This `shouldFocus` method is designed to be passed to child widgets or nodes are the value of their `focus` property.